lib/ansible/modules/cloud/vmware/vmware_vm_shell.py

```diff
@@ -141,9 +144,35 @@ def execute_command(content, vm, vm_username, vm_password, program_path, args=""
 
     creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
     cmdspec = vim.vm.guest.ProcessManager.ProgramSpec(arguments=args, envVariables=env, programPath=program_path, workingDirectory=cwd)
-    cmdpid = content.guestOperationsManager.processManager.StartProgramInGuest(vm=vm, auth=creds, spec=cmdspec)
 
-    return cmdpid
+    pm = content.guestOperationsManager.processManager
+    cmdpid = pm.StartProgramInGuest(vm=vm, auth=creds, spec=cmdspec)
+
+    if cmdpid > 0:
+        exitcode = pm.ListProcessesInGuest(vm=vm, auth=creds, pids=[cmdpid]).pop().exitCode
+
+        # If its not a numeric result code, it says None on submit
+        while (re.match('[^0-9]+', str(exitcode))):
+            time.sleep(5)
```

Comment:
Do we really want to wait for executed process ? I can imagine - if there is Windows Update or some other long running program, then playbook will run for that much amount of time, which I think is overkill.

Reply:
Hum, I think both situations are reasonable. What about to add a new argument like `wait_for_shell_exit` to determine whether wait or not?